kapp_platforms/src/windows/event_loop_windows.rs

```diff
@@ -36,6 +36,41 @@ pub unsafe extern "system" fn window_callback(
         }
         WM_KEYDOWN | WM_SYSKEYDOWN => produce_event(process_key_down(w_param, l_param)),
         WM_KEYUP | WM_SYSKEYUP => produce_event(process_key_up(w_param, l_param)),
+        WM_CHAR => {
+            let character = String::from_utf16(&[w_param as u16])
+                .unwrap()
+                .chars()
+                .next()
+                .unwrap();
+            produce_event(Event::CharacterReceived { character });
+        }
+        WM_IME_STARTCOMPOSITION => {
+            return TRUE as isize;
+        }
+        WM_IME_ENDCOMPOSITION => {
+            produce_event(Event::IMEEndComposition);
+            return TRUE as isize;
+        }
+        WM_IME_COMPOSITION => {
+            let himc = ImmGetContext(hwnd);
+            if himc == null_mut() {
+                return 0;
+            }
+            if l_param as u32 & GCS_COMPSTR != 0 {
+                let len = ImmGetCompositionStringW(himc, GCS_COMPSTR, null_mut(), 0);
+                if len == 0 {
+                    return 0;
```

Comment:
`ImmReleaseContext(hwnd, himc);` will not be called in this case, which seems like a small bug.

Also should `IMEComposition` events be sent if IMEComposition is empty?

Reply:
Nice catch! I don't think it should be sent if it's empty, although I'm not sure what the generally accepted behaviour would be.

Reply:
This new version still has the same bug if `size_bytes` == 0

It seems like `ImmReleaseContext(hwnd, himc);` could be at the end of the `WM_IME_COMPOSITION` block.

Also `buffer.set_len(size_bytes as usize / 2);` should b changed to match `  let mut buffer = Vec::<u16>::with_capacity(size_bytes as usize / std::mem::size_of::<u16>());`